src/app/icd/ICDManagementServer.cpp

```diff
@@ -82,13 +93,25 @@ Status ICDManagementServer::UnregisterClient(PersistentStorageDelegate & storage
     err = table.Remove(entry.index);
     VerifyOrReturnError(CHIP_NO_ERROR == err, InteractionModel::Status::Failure);
 
+    if (table.IsEmpty())
+    {
+        TableIsEmptyForFabric();
+    }
+
     return InteractionModel::Status::Success;
 }
 
 Status ICDManagementServer::StayActiveRequest(FabricIndex fabric_index)
 {
     // TODO: Implementent stay awake logic for end device
+    // https://github.com/project-chip/connectedhomeip/issues/24259
+    app::ICDNotifier::GetInstance().BroadcastICDManagementEvent(app::ICDListener::ICDManagementEvents::kStayActiveRequestReceived);
     return InteractionModel::Status::UnsupportedCommand;
 }
 
+void ICDManagementServer::TableIsEmptyForFabric()
+{
+    app::ICDNotifier::GetInstance().BroadcastICDManagementEvent(app::ICDListener::ICDManagementEvents::kTableUpdated);
```

Comment:
Shouldn't the ICD manager be notified only when the table is empty for all fabric ?

Reply:
It could be done that way but would require giving access to the `FabricTable`